### DISCUSSION TITLE
Wiki.js - GitHub Storage (backup) - not working - need help

### DISCUSSION THREAD

**Discussion by Frixs** [requarks/wiki#7007] [Help / Questions]
Any idea what is wrong with my WIki.js?

GitHub repo is already initialized, branch exists.

Followed this official guide: https://docs.requarks.io/storage/git

SSH keys are generated using:
`ssh-keygen -t rsa -b 4096 -C "my@github.email"`

I added the pub key to my GitHub account and I tried to add it as a deploy key directly to the repo. However, nothing works. I appreciate any help!

![01](https://github.com/requarks/wiki/assets/17912365/062d1da1-0335-4152-93fa-3f34e9c4b809)

However, this sequence of commands works (in the container of Wiki.js):

![02](https://github.com/requarks/wiki/assets/17912365/61cd58a8-2c08-45b0-a38a-cc66f977c992)

**Comment by fnafox** [comment]
Can the user wikijs is running as access (read) the key file?

**Reply by Frixs** [reply]
I cannot understand what you mean.
For clarification, here is the docker-compose file. It runs in Portainer hosted on my Synology NAS.
```docker
version: "3.9"
services:
  db:
    image: postgres:latest
    container_name: Wiki.js-DB
    hostname: wiki-js-db
    mem_limit: 512m
    cpu_shares: 768
    security_opt:
      - no-new-privileges:true
    user: 1024:101
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "wikijs", "-U", "wikiuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/wikijs/db:/var/lib/postgresql/data:rw
    networks:
      - project_bridge
    environment:
      POSTGRES_DB: wikijs
      POSTGRES_USER: wikiuser
      POSTGRES_PASSWORD: wikipass
    restart: on-failure:5

  wiki:
    image: ghcr.io/linuxserver/wikijs:latest
    container_name: Wiki-js
    hostname: wiki-js
    mem_limit: 512m
    cpu_shares: 768
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: wget --no-verbose --tries=1 --spider http://localhost:3000
    ports:
      - 3540:3000
    volumes:
      - /volume1/docker/wikijs/config:/config:rw
      - /volume1/docker/wikijs/data:/data:rw
    networks:
      - project_bridge
    environment:
      TZ: Europe/Prague
      PUID: 1024
      PGID: 101
      DB_TYPE: postgres
      DB_HOST: wiki-js-db
      DB_PORT: 5432
      DB_NAME: wikijs
      DB_USER: wikiuser
      DB_PASS: wikipass
    restart: on-failure:5
    depends_on:
      db:
        condition: service_healthy

networks:
  project_bridge:
    driver: bridge
```
UID a GUI represents the admin account on my Synology device.

**Reply by fnafox** [reply]
The wiki node server in the container is running as the default user. You can check this by running `ps aux` inside the container shell and checking what user is executing the node server command. When you've created the ssh identity, have you given said user adequate permissions to the identity file?

However, as containers are ephemeral I strongly recommend not keeping any persistent files (such as your ssh identity file) inside them, instead create the identity file outside (just like config and data) and then mount it as a volume in the compose file.

**Reply by Frixs** [reply]
Ok, thank you for these tips. I will give it a try once I get back home today afternoon and let you know then. :-)
